src/aktualizr_secondary/update_agent_ostree.cc

```diff
@@ -0,0 +1,91 @@
+#include "update_agent_ostree.h"
+
+#include "package_manager/ostreemanager.h"
+
+// TBD: consider moving this and SotaUptaneClient::secondaryTreehubCredentials() to encapsulate them in one place that
+// is shared between IP Secondary's component
+static void extractCredentialsArchive(const std::string& archive, std::string* ca, std::string* cert, std::string* pkey,
+                                      std::string* treehub_server);
+
+bool OstreeUpdateAgent::isTargetSupported(const Uptane::Target& target) const { return target.type() == "OSTREE"; }
+
+bool OstreeUpdateAgent::getInstalledImageInfo(Uptane::InstalledImageInfo& installed_image_info) const {
+  bool result = false;
+  try {
+    installed_image_info.hash = _ostreePackMan->getCurrentHash();
+    // This is the policy on a target image name in case of ostree
+    // The policy in followed and implied in meta-updater (garage-sign/push) and the backend
+    installed_image_info.name = _targetname_prefix + installed_image_info.hash;
+    installed_image_info.len = 0;
+    result = true;
+  } catch (const std::exception& exc) {
+    LOG_ERROR << "Failed to get the currently installed revision: " << exc.what();
+  }
+  return result;
+}
+
+bool OstreeUpdateAgent::download(const Uptane::Target& target, const std::string& data) {
+  std::string treehub_server;
+  bool download_result = false;
+
+  try {
+    std::string ca, cert, pkey, server_url;
+    extractCredentialsArchive(data, &ca, &cert, &pkey, &server_url);
+    // TODO: why are qe loading this credentials at all ?
```

Comment:
I am not questioning the need in ca/cert/pkey, I don't understand why, at first, we put them into the key manager and then the ostree packman fetch them from the key manager, why not just pass ca/cert/pkey to the packman.

Reply:
Fair point. My guess is that it either came from mimicking the style of the Primary or that it's only done because it's a hack. If the Secondary is just reusing the same credentials as the Primary, it shouldn't need them at all if the Primary is doing the fetching and proxying the objects to the Secondary. But I know we aren't quite there yet.

Reply:
Yeah, another item to improve implementation, in the future, but it's a minor at the moment, IMHO.